### PR TITLE
Add reworked getting started guide for local machine.

### PR DESCRIPTION
**Pull request by arturdryomov**

### CODE REVIEW

File: docs/getting-started/local-machine.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Getting Started on Local Machine
+
+## Soft Skill Dependencies
+
+Find the person responsible for maintaining the remote machine and ask for the following.
+
+* `REMOTE_MACHINE_IP_OR_HOSTNAME` — remote machine IP or hostname. Examples: `42.42.42.42`, `remote.domain.com`.
+* `REMOTE_MACHINE_PORT` — remote machine port. Example: `42`.
```
Comment (artem-zinnatullin) [inline docs/getting-started/local-machine.md]: remote machine SSH port

File: docs/getting-started/local-machine.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+### macOS
+
+```
+$ brew tap buildfoundation/homebrew-tap
+$ brew install mainframer
+```
+
+## Configuration
+
+* Is the Mainframer already used on the project? You’ll need to create the config file.
```
Comment (artem-zinnatullin) [inline docs/getting-started/local-machine.md]: in the project

File: docs/getting-started/local-machine.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+* Is the Mainframer not used on the project? You’ll need to create the config file and ignore rules.
+
+Please refer to the [documentation](../configuration.md).
+
+## Running
+
+```
+$ mainframer echo "It works!" > success.txt
```
Comment (artem-zinnatullin) [inline docs/getting-started/local-machine.md]: `>` will pipe on local machine, not remote

Comment (artem-zinnatullin) [reply docs/getting-started/local-machine.md]: also, `!` breaks quoting, pls run your sample next time :)

Comment (artem-zinnatullin) [reply docs/getting-started/local-machine.md]: Correct version would be:

```console
mainframer 'echo "It works\!" > success.txt'
```

But I'd remove "!" to remove escaping

File: docs/getting-started/local-machine.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+## Configuration
+
+* Is the Mainframer already used on the project? You’ll need to create the config file.
+* Is the Mainframer not used on the project? You’ll need to create the config file and ignore rules.
+
+Please refer to the [documentation](../configuration.md).
+
+## Running
+
+```
```
Comment (artem-zinnatullin) [inline docs/getting-started/local-machine.md]: missing highlighting

File: docs/getting-started/local-machine.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    ```
+    $ ssh {REMOTE_MACHINE_ALIAS}
+    ```
+
+## Installation
+
+### macOS
+
+```
```
Comment (artem-zinnatullin) [inline docs/getting-started/local-machine.md]: missing highlighting

File: docs/getting-started/local-machine.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+3. Send SSH public key to a person responsible for remote machine maintenance.
+
+    ```shell
+    # macOS-specific. Linux users — godspeed.
+    $ pbcopy < ~/.ssh/{SSH_KEY_FILE_NAME}.pub
+    ```
+
+4. Once you’ve received a confirmation that the remote machine is ready for you, try the connection.
+
+    ```
```
Comment (artem-zinnatullin) [inline docs/getting-started/local-machine.md]: missing highlighting

File: docs/getting-started/local-machine.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+## Authentication
+
+1. Generate SSH key and remember the resulting file name as `{SSH_KEY_FILE_NAME}`.
+
+    ```
+    $ ssh-keygen -t rsa -b 4096 -C "{REMOTE_MACHINE_USERNAME}"
+    ```
+
+2. Append the following content to `~/.ssh/config`.
```
Comment (artem-zinnatullin) [inline docs/getting-started/local-machine.md:23]: Would be great to somehow mention that ~/.ssh/config` is not required but is recommended

File: docs/getting-started/local-machine.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+* `REMOTE_MACHINE_PORT` — remote machine port. Example: `42`.
+* `REMOTE_MACHINE_USERNAME` — remote machine user, most likely a personalized, not shared one. Example: `john_doe`.
+
+Think about an alias for the remote machine.
+
+* `REMOTE_MACHINE_ALIAS` — SSH alias for remote machine. Something like `mainframe` or `build-machine`.
+
+## Authentication
+
+1. Generate SSH key and remember the resulting file name as `{SSH_KEY_FILE_NAME}`.
```
Comment (artem-zinnatullin) [inline docs/getting-started/local-machine.md:17]: Mention that it needs to happen on local machine?

File: docs/getting-started/local-machine.md (hunk 7 above)
Comment (arturdryomov) [reply docs/getting-started/local-machine.md:23]: Actually it is, we use the alias in our config. Plus it wouldn’t harm to enforce good defaults.

File: docs/getting-started/local-machine.md (hunk 8 above)
Comment (arturdryomov) [reply docs/getting-started/local-machine.md:17]: At this step it is not even possible to do this on a remote machine.

**Comment by arturdryomov** [comment]
@artem-zinnatullin, done.

File: docs/getting-started/local-machine.md [hunk 9]
```diff
@@ -4,29 +4,29 @@
 
 Find the person responsible for maintaining the remote machine and ask for the following.
 
-* `REMOTE_MACHINE_IP_OR_HOSTNAME` — remote machine IP or hostname. Examples: `42.42.42.42`, `remote.domain.com`.
-* `REMOTE_MACHINE_PORT` — remote machine port. Example: `42`.
-* `REMOTE_MACHINE_USERNAME` — remote machine user, most likely a personalized, not shared one. Example: `john_doe`.
+* `REMOTE_MACHINE_IP_OR_HOSTNAME` — IP or hostname. Examples: `42.42.42.42`, `remote.domain.com`.
+* `REMOTE_MACHINE_SSH_PORT` — SSH port. Example: `42`.
```
Comment (artem-zinnatullin) [inline docs/getting-started/local-machine.md:8]: We should avoid confusion here and use other value, 42 is already used in IP above and some users might think these are related

File: docs/getting-started/local-machine.md (hunk 8 above)
Comment (artem-zinnatullin) [reply docs/getting-started/local-machine.md:17]: well, you can authenticate with a password in some cases :) but ok

File: docs/getting-started/local-machine.md (hunk 7 above)
Comment (artem-zinnatullin) [reply docs/getting-started/local-machine.md:23]: it is not required though, it's just a convention of `ssh` to look up alias first and then use hostname

**Review by artem-zinnatullin — APPROVED** [review]

File: docs/getting-started/local-machine.md (hunk 9 above)
Comment (arturdryomov) [reply docs/getting-started/local-machine.md:8]: Changed.

**Comment by ming13** [comment]
All right, I have no idea why I cannot merge this. GitHub says _Required statuses must pass before merging_, but builds are green and change is approved.

**Pull request merged**